File: LeetCode/Easy/0070-climbing-stairs/0070-climbing-stairs.cs
public class Solution {
    // note 2 interviewer: n=45, so one could just generate a table of all the values and return table[n]
    public int ClimbStairs(int n) {
        var prev2 = 0;
        var prev1 = 1;
        var current = 0;

        for (var i = 1; i <= n; i++ )
        {
            current = prev1 + prev2;
            prev2 = prev1;
            prev1 = current;
        }

        return current;
    }
}

File: LeetCode/Easy/0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        steps_minus_2 = 1
        steps_minus_1 = 1

        if n == 1: return steps_minus_1

        for _ in range(1, n):
            steps_minus_2, steps_minus_1 = steps_minus_1, steps_minus_1 + steps_minus_2
        
        return steps_minus_1
        